extensions/azdata/src/typings/azdata-ext.d.ts

```diff
@@ -267,14 +267,14 @@ declare module 'azdata-ext' {
 				}
 			}
 		},
-		getPath(): string,
+		getPath(): string | Promise<string>,
```

Comment:
getPath and getSemVersion are async - they'll always return a promise. So doing `string | Promise<string>` makes no sense here.

Reply:
IAzdataTool implements IAzdataApi that has versions of getPath and getSemVersion that are not async. Only the implementation in getAzdataApi in api.ts uses implementations of getPath and getSemVersion that return Promise<string> because those implementations need to await for the discovery of azdata to complete. Thus the interface needs to allow both return types.

Reply:
And why does IAzdataApi need to have non-promise return values? We only have one implementation (and I don't see why we'd never have another) so it doesn't seem to make sense for the interface to have return types that will never be used.

Reply:
If I make the interface async only then that forces the implementations in AzdataTool also to be async when they do not need to be. If someone has an instance of IAzdataTool then they should be allowed to get path and semVersion synchronously only if someone has an instance of IAzdataApi do we need the async version because IAzdataApi needs to wait of azdataTool discovery to be finished before passing any commands to that tool.

Reply:
1. Can  you give a valid scenario where we'd need to have another implementation of this interface? The main reason this exists is just so it can be exposed through the interface - I don't see us ever actually having another implementation
1. Trying to always design for the "what if" scenarios is not necessarily the most useful all the time. We can always update the interface later if needed - backwards compat for extension APIs isn't something we need to be super strict about.

Reply:
Thanks for your explanations. Changing to Promise<string>

Reply:
done